src/net.c

```diff
@@ -10,10 +10,8 @@
 
 struct network {
 	struct config_net cfg;
-	struct sa laddr;
-#ifdef HAVE_INET6
-	struct sa laddr6;
-#endif
+	struct list laddrs;           /**< List of local addresses           */
+	struct list laddrs_tmp;       /**< Temp. list of local addresses     */
```

Comment:
should we consider to skip the copy and only use this function:

```c
int net_if_apply(net_ifaddr_h *ifh, void *arg);
```

there is one usecase where the user can specify which interface to use:

```
	-n <net_if>      Specify network interface
```

Reply:
- Right, this should be possible and avoids redundancy. I will try to add a commit that removes `laddrs_tmp`.
- The option `-n` and setting `net_interface` still works. If it is set, all other interfaces are ignored. It can be the name of the interface or even an IP address.